src/taskFunctions.js

```diff
@@ -0,0 +1,101 @@
+let draggedItem = null;
+let tasks = [];
+
+export function getTasks() {
+  return tasks;
+}
+
+export function setTasks(newTasks) {
+  tasks = newTasks;
+}
+
+export function handleNewTask(taskInput, populateTodoList, getTasks, setTasks) {
+  const tasks = getTasks();
+
+  if (taskInput.value.trim() === '') return;
+
+  const highestIndex = tasks.reduce((maxIndex, task) => Math.max(maxIndex, task.index), -1);
+
+  const newTask = {
+    description: taskInput.value,
+    completed: false,
+    index: highestIndex + 1,
+  };
+
+  tasks.push(newTask);
+
+  setTasks(tasks);
+
+  taskInput.value = '';
+  populateTodoList(tasks, document.getElementById('todo-list'));
```

Comment:
- [ ] according to the project requirements All changes to the To Do List should be saved in local storage. which is not the case at the moment.